api-server/src/server/boot/challenge.js

```diff
@@ -452,35 +523,178 @@ async function backendChallengeCompleted(req, res, next) {
   });
 }
 
-async function examChallengeCompleted(req, res, next) {
-  // TODO: verify shape of exam results
-  const { user, body = {} } = req;
-  const completedChallenge = pick(body, ['id', 'examResults']);
-  completedChallenge.completedDate = Date.now();
+// TODO: send flash message keys to client so they can be i18n
+function createGenerateExam(app) {
+  const { Exam } = app.models;
 
-  try {
-    await user.getCompletedChallenges$().toPromise();
-  } catch (e) {
-    return next(e);
-  }
+  return async function generateExam(req, res, next) {
+    const {
+      user,
+      query: { challengeId }
```

Comment:
```suggestion
      params: { id }
```
To reflect the above change.